### PR TITLE
add firebase redirects to handle older paths

### PR DESCRIPTION
**Pull request by devoncarew**
- add firebase redirects to handle older paths
- related to https://github.com/dart-lang/dart-pad/issues/2794

I have not tested these myself but suspect this is in-line w/ what we'd want to do

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

**Review by parlough — COMMENTED** [review]
Thanks Devon!

This overall makes sense, but it seems we might need some adjustments either with this setup or with GoRouter as ending with `index.html` results in a "Page Not Found" route and a `GoException: no routes for location: /index.html`.

Perhaps there's a preferred way to handle this that @johnpryan is familiar with?

**Comment by johnpryan** [comment]
I made some additional changes and went ahead and staged this here: https://sketch-pad-1cebb--embed-redirects-8y20kzbs.web.app 

We should be OK to merge this since this only applies to preview.dartpad.dev right now.

@devoncarew @parlough PTAL

**Comment by devoncarew** [comment]
> I made some additional changes and went ahead and staged this here: https://sketch-pad-1cebb--embed-redirects-8y20kzbs.web.app

Thanks for the updates to this PR! changes lgtm

File: pkgs/sketch_pad/firebase.json [hunk 1]
```diff
@@ -7,6 +7,33 @@
...
+      {
+        "source": "/embed-flutter_showcase.html",
+        "destination": "/?embed=true",
+        "type": 301
+      },
+      {
+        "source": "/embed-inline.html",
+        "destination": "/?embed=true",
+        "type": 301
+      }
```
Comment (parlough) [inline pkgs/sketch_pad/firebase.json:35]: Looks like this entry is duplicated. Did you mean a different file? Or can just remove:

```suggestion
      }
```

**Review by parlough — APPROVED** [review]
Thanks @devoncarew and @johnpryan!

File: pkgs/sketch_pad/firebase.json (hunk 1 above)
Comment (devoncarew) [reply pkgs/sketch_pad/firebase.json:35]: The original issue had listed `dart`, `flutter`, `html`, `inline`, and `showcase`; I had only included a few as redirects. Perhaps the the 2nd `inline` is meant to be `html`? cc @johnpryan

Comment (johnpryan) [reply pkgs/sketch_pad/firebase.json:35]: Ah, you're right this is meant to be `html`. https://github.com/dart-lang/dart-pad/pull/2800/commits/9ab54b7ec84699f236e8bfab699944c192a24867 should fix it.

**Pull request merged**